### ISSUE TITLE
Use FileResponse to serve file in google cloud storage

### ISSUE THREAD

**Issue by AnupamaGangadhar** [tiangolo/fastapi#2775]
FileResponse() works well for local path but fails to fetch file from google cloud storage. The docs don't help either to confirm if it is supported.

### Example
Assumption: auth works fine
```Python
from fastapi import FastAPI

app = FastAPI()


@app.get("/")
def get_file():
    file_path = "gs://demo/test.txt"
    return FileResponse(file_path)
```

### Description

Does FileResponse() support reading from cloud storages

### Environment

* OS: [e.g. Linux / Windows / macOS]:macOS
* FastAPI Version [e.g. 0.3.0]:0.61.2

* Python version:3.7.4

**Comment by ycd** [comment]
> The docs don't help either to confirm if it is supported.

This is a very specific use case, i don't see a reason why this should be in the docs.

As far as i have understand, this link is a bucket address and it is not in your machine right?

```
gs://demo/test.txt
```

**Comment by insomnes** [comment]
@AnupamaGangadhar `FileResponse` internally uses `aiofiles`:
```
            async with aiofiles.open(self.path, mode="rb") as file:
                more_body = True
                while more_body:
                    chunk = await file.read(self.chunk_size)
                    more_body = len(chunk) == self.chunk_size
                    await send(
                        {
                            "type": "http.response.body",
                            "body": chunk,
                            "more_body": more_body,
                        }
                    )
```
You can try to open it with `aiofiles.open` and look for result.
If it fails, you can try to open it with some lib, that supports it and send result with [StreamingResponse](https://fastapi.tiangolo.com/advanced/custom-response/?h=stream#streamingresponse)

**Comment by ycd** [comment]
@insomnes as far as i understand, he is asking something different he is trying to open a path that is provided by Google Cloud Storage. 

@AnupamaGangadhar You can not access to the files directly, you might want to use a library like [gcsfs](https://github.com/dask/gcsfs) 

```python
import gcsfs

gcs = gcsfs.GCSFileSystem(project="project name")
gcs_file_path = "gs://bucket_name/path/to/file.png"
with gcs.open(gcs_file_path) as f:
    ...
```

**Comment by AnupamaGangadhar** [comment]
@ycd  - Thank you for the response. I am aware of gcsfs. Just needed a confirmation that when using FileResponse can I access file from remote storage like GCS or not. My understanding now is that FileResponse can be used for files on local filesystem only. I have a REST endpoint that serves csv file using FileResponse and wanted to be able to serve a file from GCS instead of local storage.

Are there any plans to be able to support remote storages in the future?

**Comment by ycd** [comment]
> Just needed a confirmation that when using FileResponse can I access file from remote storage like GCS or not

Yup, as long as it is in a remote storage, you can not use `FileResponse`. 

> My understanding now is that FileResponse can be used for files on local filesystem only

That's right.

> Are there any plans to be able to support remote storages in the future?

Probably not,  because you need to access it through HTTP or FTP etc. The use case could be an external package, but it shouldn't be a part of a web framework.

**Comment by tiangolo** [comment]
Thanks for the help here everyone! 👏  🙇 

If that solves the original problem, then you can close this issue @AnupamaGangadhar  ✔️

Anyway, `gs://` is a custom protocol by Google, that's not a file path. You need to download that file and then send it in a file response.

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
